internal/pods/watcher.go

```diff
@@ -37,16 +60,56 @@ func (w *watcher) Stop() {
 	close(w.stop)
 }
 
-func NewWatcher(client kubernetes.Interface, currentNodeName string,
-	onAddFunc func(obj interface{}),
-	onUpdateFunc func(oldObj, newObj interface{}),
-	onDeleteFunc func(obj interface{})) Watcher {
+func NewWatcher(client kubernetes.Interface) Watcher {
+	locaName, err := os.Hostname()
+	if err != nil {
+		panic(err)
+	}
+
 	return &watcher{
 		client:          client,
-		currentNodeName: currentNodeName,
-		onAddFunc:       onAddFunc,
-		onUpdateFunc:    onUpdateFunc,
-		onDeleteFunc:    onDeleteFunc,
+		currentNodeName: locaName,
+		onAddFunc:       AddFunc,
+		onUpdateFunc:    UpdateFunc,
+		onDeleteFunc:    DeleteFunc,
 		stop:            make(chan struct{}),
 	}
 }
+
+func AddFunc(obj interface{}) {
+	if pod, ok := obj.(*v1.Pod); ok {
+		if config.Mode == config.ModeIstio && !IsIstioInjectedSidecar(pod) {
+			return
+		}
+		if config.Mode == config.ModeLinkerd && !IsLinkerdInjectedSidecar(pod) {
+			return
+		}
+		log.Debugf("got pod updated %s/%s", pod.Namespace, pod.Name)
+		podHostIP := pod.Status.HostIP
+		if config.CurrentNodeIP == "" {
+			if linux.IsCurrentNodeIP(podHostIP, config.IpsFile) {
+				config.CurrentNodeIP = podHostIP
+			}
+		}
+		if podHostIP == config.CurrentNodeIP || config.IsKind {
+			_ip, _ := linux.IP2Linux(pod.Status.PodIP)
+			log.Infof("update local_pod_ips with ip: %s", pod.Status.PodIP)
+			err := config.EbpfLoadPinnedMap.Update(_ip, uint32(0), ebpf.UpdateAny)
```

Comment:
these logic should not as part of `Watcher`, meybe convert Watcher to some other name, eg: LocalIPController will be better.